### PR TITLE
feat: (sdk7) change crdt executor and serialization (adr-117 rev. 2)

### PR DESCRIPTION
**Pull request by leanmendoza**
## What does this PR change?
Please read the issue https://github.com/decentraland/sdk/issues/574

## How to test the changes?
Steps in the playground: 
1. Go to the playground with the next query params added: https://playground.decentraland.org/?&sdk-branch=feat/adr-117-revision-v2&explorer-branch=feat/sdk7-crdt-adr-117-rev-2
2. When all is ready, click on Run and it should work
3. They should work as without any parameter: https://playground.decentraland.org/?&explorer-branch=dev&sdk-branch=main

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/Services/SceneControllerService/SceneControllerServiceImpl.cs [hunk 1]
```diff
@@ -183,12 +183,18 @@ public async UniTask<CRDTSceneMessage> SendCrdt(CRDTSceneMessage request, RPCCon
...
                 {
                     sendCrdtMemoryStream.SetLength(0);
                     crdtContext.scenesOutgoingCrdts.Remove(sceneNumber);
-                    KernelBinaryMessageSerializer.Serialize(sendCrdtBinaryWriter, sceneCrdtState);
-                    sceneCrdtState.ClearOnUpdated();
+
+                    foreach (var msg in sceneCrdtOutgoing)
+                    {
+                        CRDTSerializer.Serialize(sendCrdtBinaryWriter, msg.value);
+                    }
```
Comment (pbosio) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/Services/SceneControllerService/SceneControllerServiceImpl.cs:194]:
```suggestion
                    for (int i = 0; i < sceneCrdtOutgoing.Count; i++)
                    {
                        CRDTSerializer.Serialize(sendCrdtBinaryWriter, sceneCrdtOutgoing.Pairs[i].value);
                    }
```

**Review by pbosio — APPROVED** [review]

**Review by kwssbocian — APPROVED** [review]
Feature implemented correctly.
✔️No changes in behavior for playground launched with and without query parameters.

**Review by Kinerius — APPROVED** [review]

**Review by AjimenezDCL — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs [hunk 2]
```diff
@@ -14,15 +14,15 @@ private class MessageData
...
+        public int minTimeStamp;
         public ECSComponentWriteType writeType;
     }
 
     private readonly RPCContext rpcContext;
     private readonly ISceneController sceneController;
     private readonly IReadOnlyDictionary<int, ICRDTExecutor> crdtExecutors;
 
-    private readonly Dictionary<int, CRDTProtocol> outgoingCrdt = new Dictionary<int, CRDTProtocol>(60);
+    private readonly Dictionary<int, DualKeyValueSet<int, long, CRDTMessage>> outgoingCrdt = new Dictionary<int, DualKeyValueSet<int, long, CRDTMessage>>(60);
```
[the remark this answers is not in the document]
Comment (pbosio) [reply unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs:25]: It works kind of like a [sparse set](https://manenko.com/2021/05/23/sparse-sets.html)
every element is added to a list and have a dictionary to map `id => list index` to get or remove individual elements
making this data structure under-perform a dictionary when adding a element and heavily under-perform a dictionary when removing an element, but it much faster than a dictionary when iterating through all it's elements

[the remark this answers is not in the document]
Comment (AjimenezDCL) [reply unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs:25]: You are right, I created a small benchmark for this: 
image


I guess the outgoingCrdt buffer will be big enough to make this noticeable

**Review by AjimenezDCL — APPROVED** [review]

**Review by pbosio — APPROVED** [review]

**Pull request merged**